### PR TITLE
ref(replays): change consumers to use segment_id header

### PR DESCRIPTION
**Pull request by JoshFerge**
- we renamed sequence_id to segment_id. this PR changes the consumers to look for this new header field name.

### CODE REVIEW

File: src/sentry/replays/models.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ class ReplayRecordingSegment(Model):
     project_id = BoundedBigIntegerField()
     replay_id = models.CharField(max_length=32, db_index=True)
     file_id = BoundedBigIntegerField(db_index=True)
-    # recordings have multiple segments, ordered by their sequence_id
+    # recordings have multiple segments, ordered by their segment_id
+    # TODO: rename sequence_id to segment_id
     sequence_id = BoundedIntegerField()
```
Comment (cmanallen) [inline src/sentry/replays/models.py:21]: Column needs a name change and migration.  Looks like this pull was closed: https://github.com/getsentry/sentry/pull/37090/files#diff-87b3cfe1dbf66368a35666aab7253de1ae7240288e29ebd70afea9c444bd40fe

Comment (JoshFerge) [reply src/sentry/replays/models.py:21]: hi colton, see PR description:
```
i'd like to rename the django model name (while keeping the postgres column the same as outlined [here](https://develop.sentry.dev/database-migrations/#renaming-columns), but i couldn't get the migrations to generate correctly, so added a TODO for now, and can swap that later.
```

Comment (cmanallen) [reply src/sentry/replays/models.py:21]: Are you trying to rename the column in code or in the database?  The `db_column` attribute should not generate any migrations.

Comment (JoshFerge) [reply src/sentry/replays/models.py:21]: in theory yes, it shouldn't, but it is. need to dig into that, was planning on doing it after our initial ship.

**Review by cmanallen — APPROVED** [review]

**Review by wedamija — APPROVED** [review]
Ok, I've reviewed our automated checks manually. Things look ok to me. A couple of things:
 - Can you add `replays` to `MIGRATIONS_LOCKFILE_APP_WHITELIST`. This app should be added to our migrations lockfile to prevent conflicts on master.
 - I'm not sure that having a separate set of migrations is the best idea here. We should probably have these migrations with all other sentry migrations, under the `sentry` app label. Having these elsewhere breaks some of our automated checks. We could fix these, but I'm not sure if we're gaining a lot out of separating things here. Something worth discussing with your team.

Just approving the migration here, leaving it up to your team to review the code changes.

**Comment by cmanallen** [comment]
@wedamija Re: moving migrations.  I agree.  How should we proceed then?  Is there some configuration I need to set?

**Pull request merged**